gh-104692: Include commoninstall as a prerequisite for bininstall

jefferyto opened the pull request:
This ensures that `commoninstall` is completed before `bininstall` is started when parallel builds are used (`make -j install`), and so the `python3` symlink is only installed after all standard library modules are installed.

<!-- gh-issue-number: gh-104692 -->
* Issue: gh-104692
<!-- /gh-issue-number -->

jefferyto committed:
gh-104692: Include commoninstall as a prerequisite for bininstall

This ensures that `commoninstall` is completed before `bininstall` is
started when parallel builds are used (`make -j install`), and so the
`python3` symlink is only installed after all standard library modules
are installed.

zware commented on Makefile.pre.in:
```diff
@@ -1980,7 +1980,7 @@ altbininstall: $(BUILDPYTHON) @FRAMEWORKPYTHONW@
 
 .PHONY: bininstall
-bininstall: altbininstall
+bininstall: commoninstall altbininstall
```
I think we'll need to add some commentary here, it's not clear why we're doing this when we're already depending on `altbininstall`, which is what installs the targets we're linking to here:
```suggestion
# We depend on commoninstall here to make sure the installation is already usable
# before we possibly overwrite the global 'python3' symlink to avoid causing
# problems for anything else trying to run 'python3' while we install, particularly
# if we're installing in parallel with -j.
bininstall: commoninstall altbininstall
```

jefferyto replied:
I've committed your suggestion - thanks!

jefferyto committed:
Update Makefile.pre.in

Add an explanation for why commoninstall is made a prerequisite of bininstall.

Co-authored-by: Zachary Ware <zachary.ware@gmail.com>
